subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/TestReport.java

```diff
@@ -105,8 +146,13 @@ private void addTo(Object result, ConfigurableFileCollection dirs) {
      * Sets the binary test results to use to include in the report. Each entry must point to a binary test results directory generated by a {@link Test}
      * task.
      */
+    @Deprecated
     public void setTestResultDirs(Iterable<File> testResultDirs) {
-        resultDirs = getObjectFactory().fileCollection();
+        DeprecationLogger.deprecateProperty(TestReport.class, "testResultDirs").replaceWith("testResults")
+            .willBeRemovedInGradle8()
+            .withDslReference()
+            .nagUser();
+        resultDirs.setFrom(getObjectFactory().fileCollection());
         reportOn(testResultDirs);
```

Comment:
If we keep the deprecation on reportOn, this means `setTestResultsDirs` will generate two deprecations and one won't be actionable.

Reply:
That's fair; I refactored to call `resultDirs.setFrom(testResultDirs);` instead.